src/lib/jsonParseV2/parseRegistration.cpp

```diff
@@ -51,34 +51,68 @@ static bool dataProvidedParse
   ConnectionInfo*          ciP,
   ngsiv2::DataProvided*    dataProvidedP,
   const rapidjson::Value&  dataProvided,
-  std::string*             errorStringP
+  OrionError*              oeP
 )
 {
   if (!dataProvided.IsObject())
   {
-    *errorStringP = "/dataProvided/ must be a JSON object";
+    ciP->httpStatusCode = SccBadRequest;
+    oeP->code           = SccBadRequest;
+    oeP->reasonPhrase   = "BadRequest";
+    oeP->details        = "/dataProvided/ must be a JSON object";
+
     return false;
   }
 
   if (dataProvided.ObjectEmpty())
   {
-    *errorStringP = "/dataProvided/ is empty";
+    ciP->httpStatusCode = SccBadRequest;
+    oeP->code           = SccBadRequest;
+    oeP->reasonPhrase   = "BadRequest";
+    oeP->details        = "/dataProvided/ is empty";
+
     return false;
   }
 
   if (dataProvided.HasMember("entities"))
   {
-    bool b = parseEntitiesVector(ciP, &dataProvidedP->entities, dataProvided["entities"], errorStringP);
+    bool b = parseEntitiesVector(ciP, &dataProvidedP->entities, dataProvided["entities"], &oeP->details);
     if (b == false)
     {
+      ciP->httpStatusCode = SccBadRequest;
+      oeP->code           = SccBadRequest;
+      oeP->reasonPhrase   = "BadRequest";
+
+      return false;
+    }
+  }
+
+  //
+  // If idPattern is used, it MUST be ".*", else: Error
+  //
+  for (unsigned int eIx = 0; eIx < dataProvidedP->entities.size(); ++eIx)
+  {
+    ngsiv2::EntID* eP = &dataProvidedP->entities[eIx];
+
+    if ((eP->idPattern != "") && (eP->idPattern != ".*"))
+    {
+      ciP->httpStatusCode = SccNotImplemented;
+      oeP->code           = SccNotImplemented;
+      oeP->reasonPhrase   = "NotImplemented";
```

Comment:
Don't we have this literal in the corresponding .h file (not sure the name... maybe errors.h)?

Reply:
We don't, but I can add it if you want (src/lib/common/errorMessages.h)

Reply:
In addition, the usage of `typePattern` should also result in NotImplemented.

Reply:
> We don't, but I can add it if you want (src/lib/common/errorMessages.h

Not sure... have a look to errorMessage.h and it you fell it fits it, do it. Otherwise, keep it as s.

Reply:
Fixed in 3dc4bcf83f00f7937ce79cacd25417575f5beaed  (using new constants)